core/src/test/java/com/datastax/oss/driver/internal/core/context/DefaultDriverContextTest.java

```diff
@@ -0,0 +1,87 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.context;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.datastax.oss.driver.api.core.config.DefaultDriverOption;
+import com.datastax.oss.driver.api.core.config.DriverExecutionProfile;
+import com.datastax.oss.driver.internal.core.protocol.Lz4Compressor;
+import com.datastax.oss.driver.internal.core.protocol.SnappyCompressor;
+import com.datastax.oss.protocol.internal.Compressor;
+import com.datastax.oss.protocol.internal.NoopCompressor;
+import io.netty.buffer.ByteBuf;
+import java.util.Optional;
+import org.junit.Test;
+
+public class DefaultDriverContextTest {
+
+  private DefaultDriverContext buildMockedContextWithoutDef() {
+
+    DriverExecutionProfile defaultProfile = mock(DriverExecutionProfile.class);
+    when(defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION))
+        .thenReturn(Boolean.FALSE);
+    return MockedDriverContextFactory.defaultDriverContext(Optional.of(defaultProfile));
+  }
+
+  private DefaultDriverContext buildMockedContextWithDef(String compression) {
+
+    DriverExecutionProfile defaultProfile = mock(DriverExecutionProfile.class);
+    when(defaultProfile.isDefined(DefaultDriverOption.PROTOCOL_COMPRESSION))
+        .thenReturn(Boolean.TRUE);
+    when(defaultProfile.getString(DefaultDriverOption.PROTOCOL_COMPRESSION))
+        .thenReturn(compression);
+    return MockedDriverContextFactory.defaultDriverContext(Optional.of(defaultProfile));
+  }
+
+  @Test
+  public void should_create_lz4_compressor() {
+
+    DefaultDriverContext ctx = buildMockedContextWithDef("lz4");
+    Compressor<ByteBuf> compressor = ctx.getCompressor();
+    assertThat(compressor).isNotNull();
+    assertThat(compressor).isInstanceOf(Lz4Compressor.class);
+  }
+
+  @Test
+  public void should_create_snappy_compressor() {
+
+    DefaultDriverContext ctx = buildMockedContextWithDef("snappy");
+    Compressor<ByteBuf> compressor = ctx.getCompressor();
+    assertThat(compressor).isNotNull();
+    assertThat(compressor).isInstanceOf(SnappyCompressor.class);
+  }
+
+  @Test
+  public void should_create_noop_compressor_if_undefined() {
+
+    DefaultDriverContext ctx = buildMockedContextWithoutDef();
+    Compressor<ByteBuf> compressor = ctx.getCompressor();
+    assertThat(compressor).isNotNull();
+    assertThat(compressor).isInstanceOf(NoopCompressor.class);
+  }
+
+  @Test
+  public void should_create_noop_compressor_if_defined_as_none() {
+
+    DefaultDriverContext ctx = buildMockedContextWithDef("none");
+    Compressor<ByteBuf> compressor = ctx.getCompressor();
+    assertThat(compressor).isNotNull();
+    assertThat(compressor).isInstanceOf(NoopCompressor.class);
+  }
```

Comment:
We were indirectly testing the cases above via the StartupOptionsBuilderTest but it seemed worthwhile to explicitly test this as the DefaultDriverContext level as well.  As a nice side effect the additional need for a mocked DefaultDriverContext instance (perhaps given an execution profile) encouraged the move of that logic into a discrete factory impl.